ceasiompy/CPACS2GMSH/func/exportbrep.py

```diff
@@ -59,52 +61,63 @@ def export(shape, brep_dir_path, uid):
         raise FileNotFoundError(f"Failed to export {uid}")
 
 
-def engine_export(aircraft_config, brep_dir_path, symmetric_engine):
+def engine_export(cpacs, engine, brep_dir_path, engines_cfg_file_path):
     """
     Export the engine to a brep file
 
     Parameters
     ----------
-    aircraft_config: TiGL AircraftConfiguration
-        The aircraft configuration
-    brep_dir_path (obj):
+    cpacs : CPACS object
+        CPACS object (from cpacspy)
+    engine: TiGL engine
+        Engine part to be exported
+    brep_dir_path : Path
         Path object to the directory where the brep files are saved
-    symmetric_engine: bool
-        True if a second symmetric engine is needed
+    engines_cfg_file_path : Path
+        Path object to the config file for the engines
 
 
     """
-    engines_config = aircraft_config.get_engines()
-    if engines_config:
-        nb_engine = engines_config.get_engine_count()
 
-        for k in range(1, nb_engine + 1):
+    engine_uids = []
+    engine_uid = engine.get_uid()
+    engine_uids.append(engine_uid)
+    nacelle = engine.get_nacelle()
 
-            engine = engines_config.get_engine(k)
-            nacelle = engine.get_nacelle()
+    if nacelle:
+        center_cowl = nacelle.get_center_cowl()
+        if center_cowl:
+            center_cowl_uid = center_cowl.get_uid()
+            engine_uids.append(center_cowl_uid)
+            center_cowl_shape = center_cowl.build_loft()
+            export(center_cowl_shape, brep_dir_path, center_cowl_uid)
+
+        core_cowl = nacelle.get_core_cowl()
+        if core_cowl:
+            core_cowl_uid = core_cowl.get_uid()
+            engine_uids.append(core_cowl_uid)
+            core_cowl_shape = core_cowl.build_loft()
+            export(core_cowl_shape, brep_dir_path, core_cowl_uid)
+
+        fan_cowl = nacelle.get_fan_cowl()
+        if fan_cowl:
+            fan_cowl_uid = fan_cowl.get_uid()
+            engine_uids.append(fan_cowl_uid)
+            fan_cowl_shape = fan_cowl.build_loft()
+            export(fan_cowl_shape, brep_dir_path, fan_cowl_uid)
```

Comment:
Use elif to avoid to check them all..?

Reply:
we must check them all to see if the part is in the model